packages/tools/src/tools/FreehandROISculptorTool.ts

```diff
@@ -0,0 +1,685 @@
+  };
+
+  mouseMoveCallback = (evt: EventTypes.InteractionEventType): void => {
+    if (this.mode === ToolModes.Active) {
+      this._configureToolSize(evt);
+      this._updateCursor(evt);
+    } else {
+      this._commonData.mouseLocation = undefined;
+    }
+  };
+
+  /**
+   * Updates cursor size
+   *
+   * @param evt - The event
+   */
+  _updateCursor(evt: EventTypes.InteractionEventType): void {
+    const eventData = evt.detail;
+    const element = eventData.element;
+    const config = this.configuration;
+
+    const enabledElement = getEnabledElement(element);
+    const { renderingEngine, viewport } = enabledElement;
+
+    this._commonData.viewportIdsToRender = [viewport.id];
+
+    const annotations =
+      this._filterInteractableFreehandRoiAnnotationsForElement(element);
+
+    if (!annotations?.length) {
+      return;
+    }
+
+    this._commonData.mouseLocation = eventData.currentPoints.canvas;
+
+    if (this.isActive) {
+      annotations[this._commonData.selectedIndex].highlighted = true;
+    } else {
+      const canvasCoords = eventData.currentPoints.canvas;
+      const radius = this._distanceFromPoint(
+        viewport,
+        annotations[this._commonData.selectedIndex]?.data,
+        canvasCoords
+      );
+      if (radius > 0) {
+        this.toolSize = Math.min(config.maxToolSize, radius);
+      }
+    }
+
+    triggerAnnotationRenderForViewportIds(
+      renderingEngine,
+      this._commonData.viewportIdsToRender
+    );
+  }
+
+  /**
+   * Returns interactable freehand ROI annotations
+   *
+   * @param element - The viewport element
+   */
+  _filterInteractableFreehandRoiAnnotationsForElement(element: HTMLDivElement) {
+    const config = this.configuration;
+    const enabledElement = getEnabledElement(element);
+    const { renderingEngineId, viewportId } = enabledElement;
+    let interactableAnnotation = [];
+
+    const toolGroup = ToolGroupManager.getToolGroupForViewport(
+      viewportId,
+      renderingEngineId
+    );
+
+    const toolInstance = toolGroup.getToolInstance(config.referencedToolName);
+
+    config.referencedToolNames.forEach((referencedToolName: string) => {
+      const annotation = getAnnotations(referencedToolName, element);
+      if (annotation) {
+        interactableAnnotation = [...interactableAnnotation, ...annotation];
+      }
+    });
+
+    interactableAnnotation =
+      toolInstance.filterInteractableAnnotationsForElement(
+        element,
+        interactableAnnotation
+      );
+
+    return interactableAnnotation;
+  }
+
+  _configureToolSize(evt: EventTypes.InteractionEventType): void {
+    const config = this.configuration;
+
+    if (this.toolSize && config.maxToolSize) {
+      return;
+    }
+
+    const eventData = evt.detail;
+    const element = eventData.element;
+    const minDim = Math.min(element.clientWidth, element.clientHeight);
+    const sampleArea = Math.pow(minDim / 5, 2);
+    const maxRadius = Math.pow(sampleArea / Math.PI, 0.5);
```

Comment:
Math.sqrt is about 10% faster than Math.pow, according to my tests, and I find it easier to read, and (minDim*minDim)/25 is also faster because it doesn't need the function call.

Reply:
replaced Math.pow